cmd/uplink/cmd/cp.go

```diff
@@ -83,7 +86,10 @@ func upload(ctx context.Context, src fpath.FPath, dst fpath.FPath, showProgress
 	}
 
 	var access libuplink.EncryptionAccess
-	copy(access.Key[:], []byte(cfg.Enc.Key))
+	access.Key, err = cfg.Enc.LoadKey()
```

Comment:
This method returns an error if the _encryption filepath_ is empty see https://github.com/storj/storj/pull/1782/commits/8030ae2ee8c1b06ffbc7f8b6c9398aa8ae886477#diff-060bc40979507170d2ceaad8dd024efaR69 and the reason of the change in the PR https://github.com/storj/storj/pull/1848 which points to the review conversation.

If you take a look to the last Jenkins build, the current _test-sim_ doesn't have a key filepath, so the test fail when this method is called on some of the `cmd/uplink` methods.

The current upstream behavior is to accept no encryption key but the change of making the `LoadKey` to return an error when the _encryption filepath_ is empty doesn't allow to have the same behavior.

I'm wondering what's the solution that fits better to the current code base:

1. Use a package level error variable (e.g. `var ErrEncryptionFilepathEmpty = errors.New(...`) and the callers can discern over that error and proceed without returning an error.
2. Use a [errs Class](https://godoc.org/github.com/zeebo/errs#Class) which wraps the error and the callers can discern it using its [`Has` method](https://godoc.org/github.com/zeebo/errs#Class.Has).

Regarding those 2 options, I would say, to use the option 1; I'm not a huge fan of _package variables and error values_, but it's mostly a concern for big code bases, for small ones they make the job (obviously considering that people is aware per Go conventions that they don't have to modify them).

What I don't like of the 2nd option is that it isn't that simple than the 1st one; it's more awkward from a general Go point of view and the callers must use the `Has` method, plus the class will be only used for one single thing, which it doesn't look that's the purpose of them and ultimately the class must also be assigned to a package var.

__Please reply with any other idea if you have it or with the preference based on the current code base and conventions__

Reply:
#2 has been more of a convention in our codebase for what that's worth.

Reply:
the main reason is #2 gives stack traces (basically all the same reasons the go2 xerrors proposed package does it similarly)

Reply:
At the end it isn't needed to do the discern because the upstream update of the `cmd/uplink` of using `lib/uplink` has made, that those commands which could use an encryption key or the zero value, requiring a non-zero encryption key.